jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5074,9 +5074,14 @@ public String descriptorString() {
 		}
 /*[ENDIF] JAVA_SPEC_VERSION >= 15 */
 		if (this.isArray()) {
-			return name;
+		} else if (this.isPrimitiveClass()) {
+			name = new StringBuilder(name.length() + 2).
+				append('Q').append(name).append(';').toString(); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
`//$NON-NLS-1$ //$NON-NLS-2$` is no longer required since all strings have been converted to chars: `"Q" -> 'Q'` and `"L" -> 'L'`. 
```suggestion
				append('Q').append(name).append(';').toString();
```

Reply:
Could you explain what it is and why it's there?

Reply:
https://stackoverflow.com/questions/654037/what-does-non-nls-1-mean
https://stackoverflow.com/questions/18087687/what-does-the-number-after-non-nls-mean/18087688